hyperspy/_components/exponential.py

```diff
@@ -57,3 +61,97 @@ def __init__(self, A=1., tau=1., module="numexpr", **kwargs):
         )
 
         self.isbackground = False
+
+    def estimate_parameters(self, signal, x1, x2, only_current=False):
+        """Estimate the parameters for the exponential component by splitting
+        the signal window into two regions and using their geometric means
+
+        Parameters
+        ----------
+        signal : BaseSignal instance
+        x1 : float
+            Defines the left limit of the spectral range to use for the
+            estimation.
+        x2 : float
+            Defines the right limit of the spectral range to use for the
+            estimation.
+
+        only_current : bool
+            If False estimates the parameters for the full dataset.
+
+        Returns
+        -------
+        bool
+
+        """
+        super(Exponential, self)._estimate_parameters(signal)
+        axis = signal.axes_manager.signal_axes[0]
+        i1, i2 = axis.value_range_to_indices(x1, x2)
+        if i1 + 1 == i2:
+            if i2 < axis.high_index:
+                i2 += 1
+            elif i1 > axis.low_index:
+                i1 -= 1
+        i_mid = (i1 + i2) // 2
+        x_start = axis.index2value(i1)
+        x_mid = axis.index2value(i_mid)
+        x_end = axis.index2value(i2)
+
+        if only_current is True:
+            s = signal.get_current_signal()
+        else:
+            s = signal
+
+        if s._lazy:
+            import dask.array as da
+            exp = da.exp
+            log = da.log
+        else:
+            exp = np.exp
+            log = np.log
+
+        with np.errstate(divide='raise'):
+            try:
+                # use log and exp to compute geometric mean to avoid overflow
+                a1 = s.isig[i1:i_mid].data
+                b1 = log(a1)
+                a2 = s.isig[i_mid:i2].data
+                b2 = log(a2)
+                geo_mean1 = exp(b1.sum(axis=-1)/b1.shape[-1])
```

Comment:
How likely is it for `b1.sum(axis=-1)/b1.shape[-1]` > 709? Since this will still cause `RuntimeWarning: overflow encountered in exp`.

Reply:
Indeed it may happen with very high number, but overall, it seems to be fairly unlikely.

To make it more simple, replace by:
```suggestion
                geo_mean1 = exp(b1.mean(axis=-1))
```
Same for geo_mean2

Reply:
Overflowing here would mean that the geometric mean of the signal range is greater than the float64 max value, so it shouldn't be possible without another overflow already occurring in at least one channel of the signal.  

As suggested, I've simplified it to use `mean()`.